internal/api_client/task_tracker.go

```diff
@@ -64,60 +61,60 @@ func (t *TaskTracker) WaitForTask() error {
 		case <-ticker.C:
 			task, err := t.getTask()
 			if err != nil {
-				return err
+				return errors.New(*err.Message)
 			}
 
 			t.logTask(task)
 
-			switch task.Status {
+			switch *task.Status {
 			case statusInProgress, statusInProgressUppercase, statusPending:
 				continue
 			case statusFailed, statusCancelled:
 				errorMsg := fmt.Sprintf("Task with ID = %s , Name: %q Type: %q is in state %s",
-					task.ID, task.Name, task.Type, task.Status)
+					*task.Id, *task.Name, *task.Type, *task.Status)
 				tflog.Error(t.ctx, errorMsg)
 
 				return errors.New(errorMsg)
 			default:
 				tflog.Info(t.ctx, fmt.Sprintf("Task with ID = %s is in state %s, completed at %s",
-					task.ID, task.Status, task.CompletionTimestamp))
+					*task.Id, *task.Status, *task.CompletionTimestamp))
 				return nil
 			}
 		}
 	}
 }
 
-func (t *TaskTracker) getTask() (*models.Task, error) {
-	getTaskParams := tasks.NewGetTaskParamsWithTimeout(constants.DefaultVcfApiCallTimeout).
-		WithContext(t.ctx)
-	getTaskParams.ID = t.taskId
-
-	getTaskResult, err := t.client.Tasks.GetTask(getTaskParams)
+func (t *TaskTracker) getTask() (*vcf.Task, *vcf.Error) {
+	res, _ := t.client.GetTaskWithResponse(t.ctx, t.taskId)
 
-	if err != nil {
-		return nil, err
+	if res.StatusCode() == 200 {
```

Comment:
I'd create a util that test if a response is an error response, because with this autogenerated client code the http internals are leaking everywhere.

Reply:
That's a good idea, I'll add tests for some of the common functions